app/main.py

```diff
@@ -1,3 +1,16 @@
 def cache(func):
-    # Write your code here
-    pass
+    cache_of_results = {}
+    cache_of_funcs = []
+
+    def wrapper(*args, **kwargs):
+        all_args = str(args) + str(**kwargs)
+        if func.__name__ in cache_of_funcs and all_args in cache_of_results:
```

Comment:
you don't need to use the name of the function here. Store as a key - args, and as a value - result of the function